lib/services/github.js

```diff
@@ -0,0 +1,625 @@
+ * {
+ *   'status': {'success': true}
+ * }
+ */
+exports.mkdir = function (path, request, response, next, cbk) {
+	if (!isConnected(request)){
+		cbk({success:false, code:401, message:'User not logged in yet. You need to call the \'login\' service first.'});
+	}
+	else{
+		var splitPath = getPathTokens(path);
+		var reqData = null;
+		var apiPath;
+		var apiPromise;
+		switch (splitPath.length) {
+			case 0: // Error
+				return cbk({success:false, code:400, message:'Cannot create dir with an empty name.'});
+			case 1: // Create a repo
+				apiPath = '/user/repos';
+				reqData = {name: splitPath[0]};
+				apiPromise = callAPI(request.session.gh_token, apiPath, reqData, 'POST');
+				break;
+			case 2:	// Create a branch
+				apiPromise = createBranch(request.session, splitPath[0], splitPath[1]);
+				break;
+			default: // Create a folder (with a .gitignore file in it because git doesn't track empty folder)
+				var path = splitPath.slice(2).join('/');
+
+				apiPath = '/repos/' + request.session.gh_account.login + '/' + splitPath[0] + '/contents/' + path;
+				var reqData = {
+					message: 'Create '+ path,
+					content: new Buffer('').toString('base64'),
+					branch: splitPath[1]
+				}
+				apiPromise = callAPI(request, apiPath + '/.gitignore', reqData, 'PUT');
+		}
+
+		apiPromise.then(function (res) {
+			cbk({success: true, code: 200, message: 'Folder created! ' + res.toString()});
+		})
+		.catch(function(err){
+			return cbk({success:false, code: err.statusCode, message:'Cannot create a new folder. ' + err});
+		});
+	}
+}
+
+/**
+ * Create the give file
+ * @return	an object with this attribute
+ * {
+ *   'status': {'success': true}
+ * }
+ */
+exports.put = function (path, data, request, response, next, cbk) {
+	if (!isConnected(request)){
+		cbk({success:false, code:401, message:'User not logged in yet. You need to call the \'login\' service first.'});
+	}
+	else{
+		var splitPath = getPathTokens(path);
+		if (splitPath.length < 3) {
+			return cbk({success: false, message: 'Invalid path. It must at least have 3 levels.'});
+		}
+		var filePath = splitPath.slice(2).join('/');
+		commit(request.session, splitPath[0], [{
+			path: filePath,
+			content: data,
+			mode: '100644',
+			type: 'blob'
+		}], 'Create ' + path)
+		.then(function(res){
+			cbk({success:true});
+		})
+		.catch(function(err){
+			cbk({success:false, message: err});
+		})
+
+	}
+}
+
+/**
+ * Get the give file, output its content
+ * @return	the content of the file if there is no error
+ * @return	an object with this attribute
+ * {
+ *   'status': {'success': false}
+ * }
+ */
+exports.get = function (path, request, response, next, cbk) {
+	if (!isConnected(request)){
+		cbk({success:false, code:401, message:'User not logged in yet. You need to call the \'login\' service first.'});
+	}
+	else{
+		var splitPath = getPathTokens(path);
+		if (splitPath.length < 3) {
+			return cbk({success: false, message: 'Invalid path. It must at least have 3 levels.'});
```

Comment:
Isn't that a bit long? Maybe just `This folder can only contain folders not files. Files can go in sub-folders. `